metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -1238,6 +1229,8 @@ def extract_independent_datasets(
     def get_workspace_workunit(
         self, workspace: powerbi_data_classes.Workspace
     ) -> Iterable[MetadataWorkUnit]:
+        self.mapper.processed_datasets = set()
```

Comment:
Work around the leakage of processed datasets over workspaces by setting the `processed_datasets` empty for every single workspace instance.